### ISSUE TITLE
Attach iso failed  for kvm UEFI

### ISSUE THREAD

**Issue by div8cn** [apache/cloudstack#4244]
##### ISSUE TYPE

 * Bug Report
##### COMPONENT NAME

~~~
KVM UEFI
~~~

##### CLOUDSTACK VERSION

~~~
4.14.0.0
~~~

##### OS / ENVIRONMENT

OS CentOS 7.6
Compiled against library: libvirt 4.5.0
Using library: libvirt 4.5.0
Using API: QEMU 4.5.0
Running hypervisor: QEMU 2.12.0

OVMF-20180508-6.gitee3198e672e2.el7.noarch
##### SUMMARY

When VM is running in UEFI Secure Boot mode, the controller of CDROM is SATA,
But when executing attach iso cmd, IDE controller is used
So unable to complete Attache ISO

Need to be modified to STAT controller
##### STEPS TO REPRODUCE

~~~
1.Deploy VM through template, boot type select UEFI.Boot Mode select SECURE
2.Attach iso to deploy vm
~~~

##### EXPECTED RESULTS

~~~
attache iso  to vm done
~~~

##### ACTUAL RESULTS

~~~
Failed to attach device to i-2-3148-VM: internal error: No device with bus 'ide' and target 'hdc'. cdrom and floppy device hotplug isn't supported by libvirt
~~~

**Comment by DaanHoogland** [comment]
same question @div8cn do you have resources for this?
And also does this work properly with BIOS or is the problem specific to UEFI?

**Comment by div8cn** [comment]
This is a problem only under UEFI.



This is CDROM XML under ueif

<disk type='file' device='cdrom'>
<driver name='qemu' type='raw'/>
<source file='/mnt/2b9b3aaa-10ab-3fcb-987e-e8e154d4e573/232-2-843370ef-844e-329f-a901-b71462e196f0.iso'/>
<backingStore/>
<target dev='sdc' bus='**sata**'/>
<readonly/>
<alias name='sata0-0-2'/>
<address type='drive' controller='0' bus='0' target='0' unit='2'/>
</disk>





This is CDROM XML in Bois mode

<disk type='file' device='cdrom'>
<driver name='qemu'/>
<target dev='hdc' bus='**ide**'/>
<readonly/>
<alias name='ide0-1-0'/>
<address type='drive' controller='0' bus='1' target='0' unit='0'/>
</disk>



Pay attention to their bus

**Comment by DaanHoogland** [comment]
> bus=ide if bios
i guess
so do you have resource to work on this i.e. is the milestone 4.15 or 4.16?

**Comment by div8cn** [comment]
yes  bus=ide if bios

I can do relevant tests, 4.15 or 4.16

**Comment by DaanHoogland** [comment]
I don't think i can spare dev resource for this @div8cn so I won't put a milestone on it for now.

**Comment by slavkap** [comment]
Hi all, if nobody is working on this, I have a fix for attaching/detaching an ISO and deploying a VM from an ISO (because there's also a problem with this option). I'm still testing it, and when I'm ready, I could open a PR with the fix.

**Comment by rhtyd** [comment]
Fixed by https://github.com/apache/cloudstack/pull/4773

**Comment by rhtyd** [comment]
@div8cn can you test the fix from #4773 and re-open if you're able to reproduce the issue again cc @slavkap @DaanHoogland
